tests/lib/KeyStoreIndexeddb.spec.js

```diff
@@ -0,0 +1,140 @@
+        currentKeys = await KeyStore.instance.list();
+        expect(currentKeys.length).toBe(1);
+        expect(currentKeys[0].userFriendlyAddress).not.toBe(Dummy.keyInfo[0].userFriendlyAddress);
+
+        await KeyStore.instance.remove(Dummy.keyInfo[1].userFriendlyAddress);
+        currentKeys = await KeyStore.instance.list();
+        expect(currentKeys.length).toBe(0);
+
+        // check that we can't get a removed key by address
+        const removedKeys = await Promise.all([
+            KeyStore.instance._getPlain(Dummy.keyInfo[0].userFriendlyAddress),
+            KeyStore.instance._getPlain(Dummy.keyInfo[1].userFriendlyAddress)
+        ]);
+        expect(removedKeys[0]).toBeUndefined();
+        expect(removedKeys[1]).toBeUndefined();
+
+        // readd the keys to restore the state before the test
+        await Promise.all([
+            KeyStore.instance.putPlain(Dummy.keyDatabaseEntries[0]),
+            KeyStore.instance.putPlain(Dummy.keyDatabaseEntries[1])
+        ]);
+    });
+
+    it('can add and update keys', async () => {
+        // first clear database
+        await Promise.all([
+            KeyStore.instance.remove(Dummy.keyInfo[0].userFriendlyAddress),
+            KeyStore.instance.remove(Dummy.keyInfo[1].userFriendlyAddress),
+        ]);
+
+        let currentKeys = await KeyStore.instance.list();
+        expect(currentKeys.length).toBe(0);
+
+        // add a plain key
+        await KeyStore.instance.putPlain(Dummy.keyDatabaseEntries[0]);
+        currentKeys = await KeyStore.instance.list();
+        expect(currentKeys.length).toBe(1);
+
+        // add an unencrypted key
+        const dummyKey2 = new Key(Nimiq.KeyPair.unserialize(new Nimiq.SerialBuffer(Dummy.keyPairs[1])),
+            Dummy.keyInfo[1].type);
+        await KeyStore.instance.put(dummyKey2, Dummy.encryptionPassword);
+        currentKeys = await KeyStore.instance.list();
+        expect(currentKeys.length).toBe(2);
+
+        // check that the keys have been stored correctly
+        const [plainKey1, key2] = await Promise.all([
+            KeyStore.instance._getPlain(Dummy.keyInfo[0].userFriendlyAddress),
+            KeyStore.instance.get(Dummy.keyInfo[1].userFriendlyAddress, Dummy.encryptionPassword)
+        ]);
+        expect(plainKey1).toEqual(Dummy.keyDatabaseEntries[0]);
+        expect(key2.getPublicInfo()).toEqual(Dummy.keyInfo[1]);
+        expect(Nimiq.BufferUtils.equals(key2.keyPair.serialize(), Dummy.keyPairs[1])).toBe(true);
+
+        // encryption of keys is not deterministic, thus reencrypted key bytes differ from Dummy.encryptedKeyPairs[1]
+        let plainKey2 = await KeyStore.instance._getPlain(Dummy.keyInfo[1].userFriendlyAddress);
+        expect(plainKey2.encryptedKeyPair).not.toEqual(Dummy.encryptedKeyPairs[1]);
+
+        // update the key2 database entry with the original Dummy.keyDatabaseEntries[1]
+        await KeyStore.instance.putPlain(Dummy.keyDatabaseEntries[1]);
```

Comment:
Instead of using `KeyStore.instance` everywhere, you could store it in a `keyStore` variable.